strapi/strapi
# fix: use module uid for config namespace instead of dot notation

### What does it do?

Modules such as `api::myapi` and `plugin::upload` should no longer be accessible in strapi config at 'api.myapi' but instead at `api::myapi` and `plugin::upload`

Thus to retrieve config values you will need to use `config.get('plugin::upload.myconfigval')` or `config.get('api::myapi.myconfigval')`

There is a fallback left in place specifically for plugin, so that `config.get('plugin.upload')` for example will still work, but will log a deprecation warning.

### Why is it needed?

We were merging api configs with core strapi config. which was breaking strapi api config (among other potential issues)

This is also the first part of being able to standardize all the strapi containers so that we can start offering `strapi.config(uid).get(pathWithoutNamespace)` such as `strapi.config('plugin::upload').get('someval')`

### How to test it?

- Creating a content type / api should now create a config object in `strapi.config.get('api::myapi')` instead of `strapi.config.get('api.myapi')`
- creating content types with names such as 'rest' should no longer break strapi
- accessing plugin config should also be possible with `strapi.config.get('plugin.myplugin')` but should output a deprecation message

### Breaking change?

Yes, users will need to update their code to use the new paths.

A codemod has been provided which attempts to replace all usage in config get/set/has of 'plugin.' and 'api.' prefixes (except for 'api.rest' and 'api.responses' since those are valid Strapi core config paths) with 'plugin::' and 'api::'

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/13643
